torch/nn/modules/module.py

```diff
@@ -551,6 +551,81 @@ def __repr__(self):
         tmpstr = tmpstr + ')'
         return tmpstr
 
+    def summary(self, input_size):
+        def register_hook(module):
+            def hook(module, input, output):
+                class_name = str(module.__class__).split('.')[-1].split("'")[0]
+                module_idx = len(summary)
+                m_key = '%s-%i' % (class_name, module_idx + 1)
+                summary[m_key] = OrderedDict()
+                summary[m_key]['input_shape'] = list(input[0].size())
+                summary[m_key]['input_shape'][0] = None
+                summary[m_key]['output_shape'] = list(output.size())
+                summary[m_key]['output_shape'][0] = None
+
+                params = 0
+                if hasattr(module, 'weight'):
+                    params += torch.prod(torch.LongTensor(list(module.weight.size())))
```

Comment:
Instead of this, you can use `torch.numel(module.weight)`

Reply:
Thanks for the suggestion, done :)